### PR TITLE
Create a dialog box to accept or reject the invitation to become an institution administrator

### PR DESCRIPTION
**Pull request by Luiz-FS**
**Feature/Bug description:**  Create a dialog box so that a user who has been invited to become an administrator of an institution can accept or reject this invitation.

**Solution:** Using the angular directives and layout passed by the design, I created the invitation processing dialog box to be an administrator of an institution.

screenshot from 2018-03-21 11-34-48
screenshot from 2018-03-21 11-35-04
screenshot from 2018-03-21 11-36-36

**TODO/FIXME:** Check in the 'set_admin' method of the institution model if the user is a member of the institution, same in the 'add_institution_admin' method of the user model.

### CODE REVIEW

File: backend/worker.py [hunk 1]
```diff
@@ -315,8 +334,10 @@ def save_changes(admin, new_admin, institution):
             permissions = institution.get_all_hierarchy_admin_permissions()
             institution.set_admin(new_admin.key)
             self.add_permissions(new_admin, permissions)
-            permissions_filtered = filter_permissions_to_remove(admin, permissions, institution_key)
-            self.remove_permissions(admin, permissions_filtered)
+
```
Comment (mayzabeel) [inline backend/worker.py]: documentar esse método post

File: frontend/invites/processInviteUserAdmController.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    app.controller('ProcessInviteUserAdmController', function ProcessInviteUserAdmController(key, type_of_dialog, InviteService, MessageService, AuthService, $mdDialog) {
+        var processCtrl = this;
+        processCtrl.invite = {};
+        processCtrl.type_of_dialog = type_of_dialog;
+        processCtrl.isAccepting = false;
+        processCtrl.current_user = AuthService.getCurrentUser();
+
+        processCtrl.accept = function accept() {
+            InviteService.acceptInviteUserAdm(processCtrl.invite.key).then(function success() {
+                MessageService.showToast('Convite aceito com sucesso!');
```
Comment (mayzabeel) [inline frontend/invites/processInviteUserAdmController.js]: colocar essa linha no fim do método,só por organização

File: frontend/invites/processInviteUserAdmController.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                processCtrl.current_user.institutions_admin.push(processCtrl.invite.institution_key);
+                AuthService.save();
+                processCtrl.type_of_dialog = 'accepted';
+                processCtrl.isAccepting = true;
+            });
+        };
+
+        processCtrl.reject = function reject() {
+            InviteService.rejectInviteUserAdm(processCtrl.invite.key).then(function success() {
+                MessageService.showToast('Convite recusado!');
```
Comment (mayzabeel) [inline frontend/invites/processInviteUserAdmController.js]: aqui também

File: frontend/notification/notificationDirective.js [hunk 4]
```diff
@@ -151,6 +151,34 @@
...
+                icon: "account_balance",
+                action: function(properties, notification, event) {
+                    return showDialog(properties, notification, event);
+                },
+                properties: {
+                    templateUrl: "app/invites/process_invite_user_adm.html",
+                    controller: "ProcessInviteUserAdmController",
+                    controllerAs: "processCtrl",
+                    locals: {
+                        type_of_dialog: 'accept'
```
Comment (mayzabeel) [inline frontend/notification/notificationDirective.js]: pensa em um nome melhor para esse accept e accepted tá muito próximo

File: frontend/invites/processInviteUserAdmController.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                processCtrl.close();
+            });
+        };
+
+        processCtrl.close = function close() {
+            $mdDialog.hide();
+        };
+
+        (function main() {
+            InviteService.getInvite(key).then(function success(response) {
```
Comment (mayzabeel) [inline frontend/invites/processInviteUserAdmController.js]: put it in a documented function

File: frontend/invites/inviteService.js [hunk 6]
```diff
@@ -44,6 +44,30 @@
...
+            $http.put(INVITES_URI + '/' + inviteKey + '/institution_adm').then(function success(response) {
+                deferred.resolve(response);
+            }, function error(response) {
+                deferred.reject(response);
+            });
+
+            return deferred.promise;
+        };
+
+        service.rejectInviteUserAdm = function acceptInviteUserAdm(inviteKey) {
```
Comment (tiagolimpe) [inline frontend/invites/inviteService.js]: Try use "HttpService" in this function like used in requestInvitationService.js

File: frontend/invites/inviteService.js [hunk 7]
```diff
@@ -44,6 +44,30 @@
             return deferred.promise;
         };
 
+        service.acceptInviteUserAdm = function acceptInviteUserAdm(inviteKey) {
+            var deferred = $q.defer();
+
+            $http.put(INVITES_URI + '/' + inviteKey + '/institution_adm').then(function success(response) {
```
Comment (tiagolimpe) [inline frontend/invites/inviteService.js]: Try use "HttpService" in this function like used in requestInvitationService.js

Comment (tiagolimpe) [inline frontend/invites/inviteService.js]: Try use "HttpService" in this function like used in requestInvitationService.js

File: backend/test/invite_user_adm_handler_test.py [hunk 8]
```diff
@@ -232,6 +232,132 @@ def test_put_invite_in_hierarchy(self, verify_token, enqueue_task, send_notifica
...
+
+        third_inst.set_admin(admin.key)
+        third_inst.add_member(admin)
+        third_inst.add_member(new_admin)
+        third_inst.parent_institution = second_inst.key
+
+        admin.add_institution_admin(institution.key)
+        admin.add_institution_admin(third_inst.key)
+        other_admin.add_institution_admin(second_inst.key)
+        addAdminPermission(admin, institution.key.urlsafe())
```
[the remark this answers is not in the document]
Comment (Luiz-FS) [reply backend/test/invite_user_adm_handler_test.py]: Fone

File: backend/test/invite_user_adm_handler_test.py [hunk 9]
```diff
@@ -232,6 +232,132 @@ def test_put_invite_in_hierarchy(self, verify_token, enqueue_task, send_notifica
...
+        second_inst.add_member(other_admin)
+        second_inst.parent_institution = institution.key
+        second_inst.children_institutions = [third_inst.key]
+
+        third_inst.set_admin(admin.key)
+        third_inst.add_member(admin)
+        third_inst.add_member(new_admin)
+        third_inst.parent_institution = second_inst.key
+
+        admin.add_institution_admin(institution.key)
```
[the remark this answers is not in the document]
Comment (Luiz-FS) [reply backend/test/invite_user_adm_handler_test.py]: Fix in other Pull Request

File: backend/test/invite_user_adm_handler_test.py [hunk 10]
```diff
@@ -232,6 +232,132 @@ def test_put_invite_in_hierarchy(self, verify_token, enqueue_task, send_notifica
...
+
+        admin = mocks.create_user()
+        new_admin = mocks.create_user()
+        other_admin = mocks.create_user()
+
+        institution = mocks.create_institution()
+        second_inst = mocks.create_institution()
+        third_inst = mocks.create_institution()
+
+        institution.set_admin(admin.key)
```
[the remark this answers is not in the document]
Comment (Luiz-FS) [reply backend/test/invite_user_adm_handler_test.py]: Fix in other Pull Request

**Pull request merged**